src/main/java/com/woowacourse/caffeine/application/service/NotificationService.java

```diff
@@ -0,0 +1,61 @@
+package com.woowacourse.caffeine.application.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseBodyEmitter;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+class NotificationService<T> {
+
+    private static final Logger logger = LoggerFactory.getLogger(NotificationService.class);
+
+    private Map<T, ResponseBodyEmitter> subscriptions = new HashMap<>();
+    private Map<T, Queue<String>> pendingMessages = new HashMap<>();
+
+    public ResponseBodyEmitter subscribe(final T id) {
+        ResponseBodyEmitter emitter = new SseEmitter();
+        emitter.onCompletion(remove(id));
+        emitter.onTimeout(remove(id));
+        emitter.onError((e) -> {
+            logger.error("Error in event emitter", e);
+            subscriptions.remove(id);
```

Comment:
```suggestion
            remove(id)
```
제거하는 부분을 private  remove(id) 로 표현하셨던거 같아서요